src/controllers/PartialUserFormController.php

```diff
@@ -23,76 +30,119 @@ class PartialUserFormController extends UserDefinedFormController
     private static $url_handlers = [
         '$Key/$Token' => 'partial',
     ];
-
     /**
      * @var array
      */
     private static $allowed_actions = [
         'partial',
     ];
+    /**
+     * @var PartialFormSubmission
+     */
+    protected $partialFormSubmission;
 
     /**
      * Partial form
      *
      * @param HTTPRequest $request
-     * @return \SilverStripe\ORM\FieldType\DBHTMLText|void
-     * @throws \SilverStripe\Control\HTTPResponse_Exception
+     * @return HTTPResponse|DBHTMLText|void
+     * @throws Exception
      */
     public function partial(HTTPRequest $request)
+    {
+        /** @var PartialFormSubmission $partial */
+        $partial = $this->setData($request);
+        if ($this->dataRecord->PasswordProtected &&
+            $request->getSession()->get(PasswordForm::PASSWORD_SESSION_KEY) !== $partial->ID
+        ) {
+            return $this->redirect('verify');
+        }
+
+//        $this->setFailover($this->dataRecord);
```

Comment:
??

Reply:
Failover is extremely brittle and breaks often when it tries to remove wrong methods, or methods that don't have a property set.

Reply:
But shouldn't this be in the new "setData" function (L#128) anyway after setting the $this->dataRecord? What error do you have?

Reply:
It throws about 50 notices, due to in the core, there's no check if `$method['property']` is set.

Reply:
Do you have notices suppressed maybe?

Reply:
Yeah I get it now, about 61 notices. I'll see if there's another way